go/mysql/replication_position.go

```diff
@@ -169,3 +169,31 @@ func (rp *Position) UnmarshalJSON(buf []byte) error {
 	}
 	return nil
 }
+
+// Comparable returns whether the receiver is comparable to the supplied position, based on whether one
+// of the two positions contains the other.
+func (rp *Position) Comparable(other Position) bool {
+	if !rp.GTIDSet.Contains(other.GTIDSet) && !other.GTIDSet.Contains(rp.GTIDSet) {
+		return false
+	}
+
+	return true
+}
+
+// AllPositionsComparable returns true if all positions in the supplied list are comparable with one another, and false
+// if any are non-comparable.
+func AllPositionsComparable(positions []Position) bool {
+	if len(positions) < 2 {
```

Comment:
The loop should work fine without this, shouldn't it?

Reply:
It should, but I personally like the idea of showing this at the top - that less than 2 means no comparisons are necessary.

Reply:
It's more explicit, but you're right

Reply:
I suppose it's simpler without, so I'll remove it.

Reply:
Removed.